### ISSUE TITLE
Invalid BlockControlStructureSpacing when control structures are separated by different comment types only

### ISSUE THREAD

**Issue by spaceemotion** [slevomat/coding-standard#1002]
Given the following snippet:

```php
foreach ($foo as $bar) {
    // TODO lorem ipsum
    /** @var string $baz */
    foreach ($bar as $baz) {
        echo $baz;
    }
}
```

Which gives me:

```
Expected 0 lines before "foreach", found 1.
SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing.IncorrectLinesCountBeforeFirstControlStructure
```

with the following config:

```xml
<rule ref="SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing">
  <!-- Remove case & default from the list -->
  <properties>
    <property name="tokensToCheck" type="array">
      <element value="T_IF"/>
      <element value="T_DO"/>
      <element value="T_WHILE"/>
      <element value="T_FOR"/>
      <element value="T_FOREACH"/>
      <element value="T_SWITCH"/>
      <element value="T_TRY"/>
    </property>
  </properties>
</rule>
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/e2efebfd65866c1484ec7b400e0c37f96baffd08

**Comment by spaceemotion** [comment]
Thanks for all these quick fixes 🙌 much appreciated!
